lib/helpers/results_helpers.js

```diff
@@ -257,32 +257,37 @@ module.exports = class ResultsHelpers {
             statementResults[lib][statementName].pretty = `FALSE (${rawStatementResult})`;
           }
         }
-        // create clause results for all localIds in this statement
-        const localIds = MeasureHelpers.findAllLocalIdsInStatementByName(measure, lib, statementName);
-        for (const localId in localIds) {
-          const clause = localIds[localId];
-          const clauseResult = {
-            // if this clause is an alias or a usage of alias it will get the raw result from the sourceLocalId.
-            raw:
-              rawClauseResults[lib] != null
-                ? rawClauseResults[lib][clause.sourceLocalId != null ? clause.sourceLocalId : localId]
-                : undefined,
-            statementName,
-          };
-          clauseResult.final = this.setFinalResults({
-            statementRelevance,
-            statementName,
-            rawClauseResults,
-            lib,
-            localId,
-            clause,
-            rawResult: clauseResult.raw,
-          });
-          clauseResults[lib][localId] = clauseResult;
+
+        if (includeClauseResults) {
+          // create clause results for all localIds in this statement
+          const localIds = MeasureHelpers.findAllLocalIdsInStatementByName(measure, lib, statementName);
+          for (const localId in localIds) {
+            const clause = localIds[localId];
+            const clauseResult = {
+              // if this clause is an alias or a usage of alias it will get the raw result from the sourceLocalId.
+              raw:
+                rawClauseResults[lib] != null
+                  ? rawClauseResults[lib][clause.sourceLocalId != null ? clause.sourceLocalId : localId]
+                  : undefined,
+              statementName,
+            };
+            clauseResult.final = this.setFinalResults({
+              statementRelevance,
+              statementName,
+              rawClauseResults,
+              lib,
+              localId,
+              clause,
+              rawResult: clauseResult.raw,
+            });
+            clauseResults[lib][localId] = clauseResult;
+          }
         }
       }
     }
-    return { statement_results: statementResults, clause_results: clauseResults };
+    const response = { statement_results: statementResults };
+    response.clause_results = includeClauseResults ? clauseResults : 'These will be included if you set includeClauseResults to true in the request options.';
```

Comment:
The value here should be undefined or null instead of 'These will be included....' string